core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -62,6 +62,10 @@ class SnappySession(@transient private val sc: SparkContext,
     this(sc, None)
   }
 
+  def this(spark: SparkSession) {
+    this(spark.sparkContext, None)
+  }
+
```

Comment:
I don't think this constructor makes sense. It gives the impression to users that this will somehow be tied to the passed SparkSession while this creates a new one. Let users pass SparkContext explicitly so the second constructor which is private[sql] should now be public.